### ISSUE TITLE
[Event Parameter Request] codeunit 5812 "Calculate Standard Cost" - OnAfterCalcRtngLineCost

### ISSUE THREAD

**Issue by Duizy1971** [microsoft/ALAppExtensions#25599]
### Describe the request

    Add variables ParentItem and CostTime to event procedure OnAfterCalcRtngLineCost:
        
    procedure CalcRtngLineCost(RoutingLine: Record "Routing Line"; MfgItemQtyBase: Decimal; var SLCap: Decimal; var SLSub: Decimal; var SLCapOvhd: Decimal; var ParentItem: Record Item)
    ....

        OnAfterCalcRtngLineCost(RoutingLine, MfgItemQtyBase, SLCap, SLSub, SLCapOvhd, StdCostWkshName, ParentItem, CostTime);
    end;
    
    [IntegrationEvent(false, false)]
    local procedure OnAfterCalcRtngLineCost(RoutingLine: Record "Routing Line"; MfgItemQtyBase: Decimal; var SLCap: Decimal; var SLSub: Decimal; var SLCapOvhd: Decimal; var StdCostWkshName: Text[50]; var ParentItem: Record Item; CostTime: Decimal)
    begin
    end;

### Additional context

This issue goes together with https://github.com/microsoft/ALAppExtensions/issues/25598<br /><br />Internal work item: AB#494740

**Comment by AleksandricMarko** [comment]
Hi @Duizy1971 
ParentItem does not exist as part of procedure CalcRtngLineCost
Please review the request.

**Comment by Duizy1971** [comment]
https://github.com/microsoft/ALAppExtensions/issues/25597 is completed by Jesper so it should be possible to implement this issue.

**Comment by Duizy1971** [comment]
Hello, We are in dire need of this event to get out customer from BC14 to BC23.04 and not much later. Can you please pick up this issue?

**Comment by AleksandricMarko** [comment]
Hello @Duizy1971 
Variable ParentItem is part of procedure CalcRtngCost, but still value is not available in global procedure CalcRtngLineCos.
Is the option for you to add OnCalcRtngCostOnAfterCalcRtngLineCost(desired parameters); as a part of loop?

**Comment by Duizy1971** [comment]
Sorry Aleksandric,

Issue https://github.com/microsoft/ALAppExtensions/issues/25598
Must be done first

**Comment by AleksandricMarko** [comment]
Hi @Duizy1971 
Ok, then, this one will wait for other change.

**Comment by Duizy1971** [comment]
Hi Marko,

Can I ask then what is happening to 'AB#495832'  then?
Jesper should have done all the prepwork by now as far as I can tell from build 15733.


Met vriendelijke groet / Kind regards,

Marcel Duizendstraal

Senior Software Engineer

T.



+31 6 34204770

M.



***@***.***

[https://storage.letsignit.com/5fe9d9be947aea000a6a7a84/generated/effects_62ca5699e17dd8b5c8582602759c111e6066dc25c39ebf5ac094b302.png]

[https://storage.letsignit.com/5fe9d9be947aea000a6a7a84/generated/effects_3517cbd873ba9e9adef951274fd8fe30d37bd7125d6946a22180a282.png]<https://cloud.letsignit.com/collect/b/659bbca337211a3644816d08?p=xSKbv52R9rh-TBlo1SmhTkgGC6v2koBT4IPk9ls43jnTThQSm4Dan5f7QTee1NZCnoYUjmBgx5vCv0vg3WTnBk-RmrSw4VzCZapWxLUYcGJjuFm-RXkpuGnSTghoXcubuNcx5c8_uet-QBmFjdQwWNIVeje8_g-tSgqSdCAEa31TfP8Ykk-nbLTA-aMhMMpf>


________________________________

**Comment by JesperSchulz** [comment]
Internal work item 494740 was closed 08-05-2024. I expect that this has been resolved? If not, kindly create a new issue and we'll be on it!
